app/services/GetMovementService.scala

```diff
@@ -31,18 +32,20 @@ class GetMovementService @Inject()(getMovementConnector: GetMovementConnector,
                                    getWineOperationsService: GetWineOperationsService,
                                    getMovementHistoryEventsService: GetMovementHistoryEventsService)(implicit ec: ExecutionContext) {
 
-  def getRawMovement(ern: String, arc: String)(implicit hc: HeaderCarrier) : Future[GetMovementResponse] =
-    getMovementConnector.getMovement(ern, arc).map {
+  def getRawMovement(ern: String, arc: String, sequenceNumber: Option[Int] = None)(implicit hc: HeaderCarrier) : Future[GetMovementResponse] =
+    getMovementConnector.getMovement(ern, arc, sequenceNumber).map {
       case Right(movement) => movement
       case Left(errorResponse) =>
         throw MovementException(s"Failed to retrieve movement from emcs-tfe: $errorResponse")
 
     }
 
-  def getMovement(ern: String, arc: String)(implicit hc: HeaderCarrier): Future[GetMovementResponse] =
-    getRawMovement(ern, arc).flatMap { movement =>
+  def getMovement(ern: String, arc: String, sequenceNumber: Option[Int] = None, historyEvents: Option[Seq[MovementHistoryEvent]] = None)
```

Comment:
Not sure what's going on here, please could you clear some stuff up for me?

- This is only called in one place outside of tests, on line 71, with `Some(historyEvents)`. Does it need to be an Option with a default value of `None`?
- In the for comprehension, a `historyEvents` variable is created, using `historyEvents` from the function parameter. This variable is used to call `getMovementHistoryEvents`, which has already been called on line 65. Does this need to happen?

What I'm getting at is, could this be rewritten like this:
```scala
  def getMovement(ern: String, arc: String, sequenceNumber: Option[Int] = None, historyEvents: Seq[MovementHistoryEvent])
                 (implicit hc: HeaderCarrier): Future[GetMovementResponse] =

    getRawMovement(ern, arc, sequenceNumber).flatMap { movement =>
      for {
        itemsWithWineOperations <- getWineOperationsService.getWineOperations(movement.items)
        itemsWithWineAndPackaging <- getPackagingTypesService.getMovementItemsWithPackagingTypes(itemsWithWineOperations)
        itemsWithCnCodeInfo <- getCnCodeInformationService.getCnCodeInformation(itemsWithWineAndPackaging)
        itemsWithWineAndPackagingAndCnCodeInfo = itemsWithCnCodeInfo.map {
          case (item, cnCodeInfo) => item.copy(
            unitOfMeasure = Some(cnCodeInfo.unitOfMeasure),
            productCodeDescription = Some(cnCodeInfo.exciseProductCodeDescription)
          )
        }
      } yield movement.copy(
        items = itemsWithWineAndPackagingAndCnCodeInfo,
        eventHistorySummary = Some(historyEvents)
      )
    }
```

or is this done intentionally?

Reply:
Line 48 uses a fold to either get the movement history events (if they have not been supplied) or uses the value passed in the function parameter.

`historyEvents <- historyEvents.fold(getMovementHistoryEventsService.getMovementHistoryEvents(ern, arc))(history => Future.successful(history))`